packages/chart/src/XYAxis.ts

```diff
@@ -183,23 +192,37 @@ export class XYAxis extends SVGWidget {
         let selected = [];
         const currSel: any = d3BrushSelection(this.svgBrush.node());
         if (currSel) {
-            selected = this.data().filter(function (d) {
-                let pos = d[0];
-                pos = this.dataPos(pos) + this.domainAxis.bandwidth() / 2;
-                return pos >= currSel[0] && pos <= currSel[1];
-            }, this);
+            if (this.selectionMode2D()) {
+                selected = this.data().filter(function (d) {
```

Comment:
Minor:  I would probably put the `if (this.selectionMode2D()) {` inside the `selected = this.data().filter(function (d) {` and reuse some of the "pos calculations.